MineInAbyss/src/main/java/com/derongan/minecraft/mineinabyss/Ascension/Effect/Effects/AbstractAscensionEffect.java

```diff
@@ -15,22 +15,28 @@
     int strength;
     int offset;
     int iterations;
+    private int iterationsRemaining;
 
     AbstractAscensionEffect(int offset, int strength, int durationRemaining, int iterations) {
         this.iterations = iterations;
         this.durationRemaining = durationRemaining;
         this.elapsed = 0;
         this.offset = offset;
         this.strength = strength;
+        this.iterationsRemaining = iterations;
     }
 
     @Override
     public void applyEffect(Player player, int ticks) {
-        for (int iterationsScheduled = 0; iterationsScheduled != iterations; iterationsScheduled++) {
-            Bukkit.getScheduler().runTaskLater(JavaPlugin.getPlugin(MineInAbyss.class), () -> {
-                applyEffect(player);
-            }, (offset + iterationsScheduled + (durationRemaining * iterationsScheduled)));
-        }
+        Bukkit.getScheduler().runTaskLater(JavaPlugin.getPlugin(MineInAbyss.class), () -> {
+            applyEffect(player);
```

Comment:
This looks a bit weird. You are recursively calling applyEffect, that is fune, but you are also recursively calling applyEffect and repeatEffect? At a quick glance it looks like this recursion is going to grow exponentially

Reply:
That is not the actual recursion.
Forgot an if statement.